src/sentry_plugins/bitbucket/endpoints/webhook.py

```diff
@@ -0,0 +1,225 @@
+
+# from sentry_plugins.exceptions import ApiError
+# from sentry_plugins.bitbucket.client import BitbucketClient
+
+logger = logging.getLogger('sentry.webhooks')
+
+
+def is_anonymous_email(email):
+    # todo(maxbittker)
+    return email[-25:] == '@users.noreply.bitbucket.com'
+
+
+def get_external_id(username):
+    # todo(maxbittker)
+    return 'bitbucket:%s' % username
+
+
+class Webhook(object):
+    def __call__(self, organization, event):
+        raise NotImplementedError
+
+
+def parse_raw_user(raw):
+    # captures content between angle brackets
+    return re.search('(?<=<).*(?=>$)', raw).group(0)
+
+
+class PushEventWebhook(Webhook):
+    # https://confluence.atlassian.com/bitbucket/event-payloads-740262817.html#EventPayloads-Push
+    def __call__(self, organization, event):
+        authors = {}
+
+        # client = BitbucketClient()
+        try:
+            repo = Repository.objects.get(
+                organization_id=organization.id,
+                provider='bitbucket',
+                external_id=six.text_type(event['repository']['uuid']),
+            )
+        except Repository.DoesNotExist:
+            raise Http404()
+
+        for change in event['push']['changes']:
+            for commit in change['commits']:
+                if RepositoryProvider.should_ignore_commit(commit['message']):
+                    continue
+
+                author_email = parse_raw_user(commit['author']['raw'])
+                if '@' not in author_email:
+                    author_email = u'{}@localhost'.format(author_email[:65])
+
+                # TODO(dcramer): we need to deal with bad values here, but since
+                # its optional, lets just throw it out for now
+                if len(author_email) > 75:
+                    author = None
+                elif author_email not in authors:
+                    authors[author_email] = author = CommitAuthor.objects.get_or_create(
+                        organization_id=organization.id,
+                        email=author_email,
+                        defaults={
+                            'name': commit['author']['user']['display_name'][:128],
+                        }
+                    )[0]
+                else:
+                    author = authors['author_email']
+                try:
+                    with transaction.atomic():
+
+                        Commit.objects.create(
+                            repository_id=repo.id,
+                            organization_id=organization.id,
+                            key=commit['hash'],
+                            message=commit['message'],
+                            author=author,
+                            date_added=dateutil.parser.parse(
+                                commit['date'],
+                            ).astimezone(timezone.utc),
+                        )
+
+                        # TODO(maxbittker) can't make these work until i save the auth token somewhere
```

Comment:
i think we kind of discussed this in slack, but we should try to just handle this part when backfilling the commits in `Release.set_commits` in the sentry repo. If we do it there, we will already be making requests to bitbucket to determine which commits are part of the release, and we'll can use the requesting user's auth.